torch/_ops.py

```diff
@@ -782,6 +775,85 @@ def tags(self):
     # TODO: add more methods to expose information about input and output arguments
 
 
+# TorchBindOpOverload are those custom ops which have at least one overload's
+# schema consists of torch.ScriptObject (i.e. custom class) input.
+# TorchBindOpOverload will skip C++ dispatcher and purely dispatched in python
+# when its inputs contain FakeScriptObject in a similar way as higher order ops.
+class TorchBindOpOverload(OpOverload):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _fallthrough_keys(self) -> List[DispatchKey]:
+        _DEFAULT_FALLTHROUGH_KEYS = [
+            DispatchKey.Autograd,
+            DispatchKey.AutogradCPU,
+            DispatchKey.AutogradCUDA,
+            DispatchKey.ADInplaceOrView,
+            DispatchKey.PythonTLSSnapshot,
+        ]
+        return [
+            key
+            for key in _DEFAULT_FALLTHROUGH_KEYS
+            if key not in self.py_kernels
+            or self.py_kernels[key] is torch.library.fallthrough_kernel
+        ]
```

Comment:
This logic should be:
- if the op doesn't have a py_kernel or a C++ implementation for one of these keys, then treat it as a fallthrough.

Reply:
Also, name this to something like "mayUseFallthroughInsteadOfFallback" and add a big TODO. The problem is that we should be calling the fallback for these, but a fallthrough is almost close enough to the fallthrough in most cases that we care about that we can use a fallthrough